### PR TITLE
4196: updated readme with managed provider and new imports

### PR DESCRIPTION
**Pull request by mrand-phunware**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,39 +124,73 @@ zoneConfigurations.add(zoneConfiguration);
...
-// Create PwLocationProviderFactory
-PwLocationProviderFactory pwLocationProviderFactory = new PwLocationProviderFactory();
+// Create PwMseLocationProvider
+PwLocationProvider locationProvider = CMXProviderFactory.create(mContext, <!-- YOUR_VENUE_GUID -->)
+.createLocationProvider();
+```
 
+If for some reason multiple copies of  your building exist in MaaS, only the first one registered with CMX will report the correct floorIDs. For any other building copies, you will need to pass a map of MSE floor IDs to MaaS floor IDs like so:
+```java
+// Create floor ID mappin
```
Comment (astolzberg) [inline README.md]: mapping*

File: README.md [hunk 2]
```diff
@@ -139,39 +124,73 @@ zoneConfigurations.add(zoneConfiguration);
...
+        || mseFloorIds.length == 0) {
+    Log.d(TAG, "The floorIdMapping length doesn't match");
+    return null;
+}
+
+int length = cmxFloorIds.length;
+for (int i = 0; i < length; i++) {
+    floorIdMap.put(String.valueOf(cmxFloorIds[i]), (long) pwFloorIDs[i]);
+}
 // Create PwMseLocationProvider
```
Comment (astolzberg) [inline README.md]: Mse-->CMX

File: README.md [hunk 3]
```diff
@@ -139,39 +124,73 @@ zoneConfigurations.add(zoneConfiguration);
...
   }
 };
-
 // Create PwMockLocationProvider
-PwMockLocationProvider locationProvider = (PwMockLocationProvider) pwLocationProviderFactory.getPwMockLocationProvider(mContext, jsonFileName, listener, repeat);
+PwMockLocationProvider locationProvider = (PwMockLocationProvider) MockProviderFactory.create(mContext, appPackageName, jsonFileName, listener);
+```
+
+####PwManagedLocationProvider
+The PwManagedLocationProvider class allows you to implement Phunware's managed location software to further improve upon locations from your location provider hardware. Currently supported location providers are: CMX, CMX + Senion, Senion, and BeaconPoint. From a code standpoint, setting up a managed location provider is the same regardless of your setup. All of these attributes are set through the MaaS portal.
```
Comment (astolzberg) [inline README.md]: should you mention that some id's like `bpOrgId` are only required if you have that technology enabled?

File: README.md [hunk 4]
```diff
@@ -139,39 +124,73 @@ zoneConfigurations.add(zoneConfiguration);
...
 ```
 
-####PwMseLocationProvider
-The PwMseLocationProvider class defines the interface for configuring the delivery of Cisco Mobility Services Engine (MSE) location-related events to your application. You use an instance of this class to establish the parameters that determine when location events should be delivered and to start and stop the actual delivery of those events.
+####PwCmxLocationProvider
+The PwCmxLocationProvider class defines the interface for configuring the delivery of Cisco Mobility Services Engine (CMX) location-related events to your application. You use an instance of this class to establish the parameters that determine when location events should be delivered and to start and stop the actual delivery of those events.
 ```java
-// Create PwLocationProviderFactory
-PwLocationProviderFactory pwLocationProviderFactory = new PwLocationProviderFactory();
+// Create PwMseLocationProvider
```
Comment (astolzberg) [inline README.md]: mse-->cmx

File: README.md [hunk 5]
```diff
@@ -139,39 +124,74 @@ zoneConfigurations.add(zoneConfiguration);
...
   }
 };
-
 // Create PwMockLocationProvider
-PwMockLocationProvider locationProvider = (PwMockLocationProvider) pwLocationProviderFactory.getPwMockLocationProvider(mContext, jsonFileName, listener, repeat);
+PwMockLocationProvider locationProvider = (PwMockLocationProvider) MockProviderFactory.create(mContext, appPackageName, jsonFileName, listener);
+```
+
+####PwManagedLocationProvider
+The PwManagedLocationProvider class allows you to implement Phunware's managed location software to further improve upon locations from your location provider hardware. Currently supported location providers are: CMX, CMX + Senion, Senion, and BeaconPoint. From a code standpoint, setting up a managed location provider is the same regardless of your setup. All of these attributes are set through the MaaS portal. Some of the IDs passed into the builder can be null if those location providers are not being used.
```
Comment (mrand-phunware) [inline README.md]: clarify that parts of the builder pattern can be excluded if not used

File: README.md [hunk 6]
```diff
@@ -139,39 +124,74 @@ zoneConfigurations.add(zoneConfiguration);
...
+
+```java
+// Create ManagedLocationProviderFactoryBuilder
+ManagedProviderFactory.ManagedProviderFactoryBuilder builder =
+new ManagedProviderFactory.ManagedProviderFactoryBuilder();
+
+//build location provider
+PwManagedLocationProvider provider = builder.application(mApplication)
+      .context(mContext)
+      .venueId(<!--YOUR_BUILDING_ID-->)
```
Comment (mrand-phunware) [inline README.md]: change to building id

**Comment by astolzberg** [comment]
Looks good!

**Pull request merged**